python/adjoint/filters.py

```diff
@@ -7,6 +7,19 @@
 import meep as mp
 from scipy import special
 
+def compute_mg_dims(Lx,Ly,resolution):
+    '''Compute the material grid dimensions from
+    the corresponding resolution, x-size, and y-size.
+    The grid dimensions must be odd.    
```

Comment:
Why?

Reply:
If the number of "taps" is not odd, it's no longer a zero-phase filter.